src/runtime/stopwatch.ts

```diff
@@ -0,0 +1,81 @@
+import { messages } from '@cucumber/messages'
+import { stopwatch, Stopwatch, duration, Duration } from 'durations'
+
+export interface ITestRunStopwatch {
+  from: (duration: Duration) => ITestRunStopwatch
+  start: () => ITestRunStopwatch
+  stop: () => ITestRunStopwatch
+  duration: () => Duration
+  timestamp: () => messages.ITimestamp
+}
+
+export class RealTestRunStopwatch implements ITestRunStopwatch {
+  private readonly stopwatch: Stopwatch = stopwatch()
+  private base: Duration = null
+
+  from(duration: Duration): ITestRunStopwatch {
+    this.base = duration
+    return this
+  }
+
+  start(): ITestRunStopwatch {
+    this.stopwatch.start()
+    return this
+  }
+
+  stop(): ITestRunStopwatch {
+    this.stopwatch.stop()
+    return this
+  }
+
+  duration(): Duration {
+    const current = this.stopwatch.duration()
+    if (this.base !== null) {
+      return duration(this.base.nanos() + current.nanos())
```

Comment:
I'm confused by this. What exactly is happening here? Could we add some tests for this? (maybe allow stopwatch to be injected if mocking it out is hard)

Reply:
Good point will add some tests.

The `base` thing here is used in the parallel runtime. We have a stopwatch at the test run level, but we can't pass it from the coordinator to the workers as it's not serializable, so we instead pass the current elapsed duration in nanos and then seed it to the `base` on the stopwatch that we init in the worker.

Reply:
Not seeing any tests for this.